src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static Task[] allTasks;
+    private static final int TASK_CAPACITY = 100;
+    private static int count;
+
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_UNMARK = "unmark";
+
+    private static final String MESSAGE_ADD = "Got it. I've added this task:";
+    private static final String MESSAGE_EMPTY_LIST = "Your list is empty.";
+    private static final String MESSAGE_ERROR = "Sorry, I did not understand that.";
+    private static final String MESSAGE_GOODBYE = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_LIST_HEADER = "Here are the tasks in your list:";
+    private static final String MESSAGE_MARK = "Nice! I've marked this task as done:";
+    private static final String MESSAGE_UNMARK = "OK, I've marked this task as not done yet:";
+    private static final String MESSAGE_WELCOME = "Hello! I'm FredBot.\nWhat can I do for you?";
+
+    private static final String PREFIX_BY = "/by";
+    private static final String PREFIX_FROM = "/from";
+    private static final String PREFIX_TO = "/to";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        initTaskList();
+        showWelcomeMessage();
+        while (true) {
+            String command = readUserInput();
+            executeCommand(command);
+        }
+    }
+
+    private static void initTaskList() {
+        allTasks = new Task[TASK_CAPACITY];
+        count = 0;
+    }
+
+    private static void showWelcomeMessage() {
```

Comment:
Should the `static` keyword be removed? there is no need to declare `private` methods as `static` since `private` means they can only be accessed within the class, hence never have a chance to be called outside the class like a public static method